src/App.js

```diff
@@ -0,0 +1,178 @@
+    }
+
+    let onFilterCategory = async (catergory) => {
+        let url = `https://api.themoviedb.org/3/movie/${catergory}?api_key=${apiKey}&language=en-US&page=1`
+        let data = await fetch(url)
+        let dataResult = await data.json()
+        console.log(dataResult)
+        setMovies(dataResult.results)
+        setcategoryLoadMore(catergory)
+    }
+
+    let onShowVideo = async (movieId) => {
+        let url = `https://api.themoviedb.org/3/movie/${movieId}/videos?api_key=${apiKey}&language=en-US;`
+        let data = await fetch(url)
+        let respones = await data.json()
+        setModal(true)
+        if (respones.results.length > 0) {
+            setMovieVideo(respones.results[0].key)
+        }
+        console.log(respones)
+    }
+
+    let onSearch = (keyword) => {
+        if (keyword === '') {
+            setMovies(movieList)
+        } else {
+            setMovies(movies.filter((movie) => movie.title.toLowerCase().includes(keyword.toLowerCase())))
+        }
+    }
+
+    let onSortByHighestPopularity = () => {
+        let newMovieList = [...movies].sort((a, b) => a.popularity > b.popularity ? -1 : 1)
+        setMovies(newMovieList)
+    }
+
+    let onSortByLowestPopularity = () => {
+        let newMovieList = [...movies].sort((a, b) => a.popularity < b.popularity ? -1 : 1)
+        setMovies(newMovieList)
+    }
+
+    let onFilterRating = (rating) => {
+        console.log(rating)
+        console.log(rating.min, rating.max)
+        setMovies(movieList.filter((movie) => movie.vote_average >= rating.min && movie.vote_average <= rating.max))
+        console.log('rating movie', movies)
+    }
+
+    let handlePageChange = async (pageNumber) => {
+        setPageNumber(pageNumber)
+        let url = `https://api.themoviedb.org/3/movie/${categoryLoadMore}?api_key=${apiKey}&language=en-US&page=${pageNumber}`
+        let data = await fetch(url)
+        let dataResult = await data.json()
+        setMovies(dataResult.results)
+    }
+
+    useEffect(() => {
+        currentPlaying()
+        fetchGenres()
+    }, [])
+
+    // useEffect(() => {
+    //     currentPlaying()
+    // }, [genres]) // put genres inside array mean. hey, if genres change exicute function currentPlaying
+
+
+    if (movies === []) {
+        return (
+            <div>
+                Loading...
+            </div>
+        )
+    }
+    return (
+        <div className="App">
+            <NavbarMovie 
+                onSearch={onSearch}
+                onFilterCategory={onFilterCategory}
+                onSortByHighestPopularity={onSortByHighestPopularity}
+                onSortByLowestPopularity={onSortByLowestPopularity}
+            />
+            <div className="row w-100" style={{ paddingTop: '100px' }}>
+                <div className="col-lg-1"></div>
+                <div className="col-lg-10 banner">
+                    <Banner />
+                </div>
+                <div className="col-lg-1"></div>
```

Comment:
so instead of having extra columns like this, you can also do col-offset https://getbootstrap.com/docs/4.0/layout/grid/